docx_editor/session.py

```
@@ -180,10 +181,17 @@ def _abort(message: str) -> SessionError:
         time.sleep(0.1)
 
-    kc = _client(connection_file)
+    kc = None
     try:
+        # Inside the guard: the kernel writes the connection file
+        # non-atomically, so a mid-write read fails to parse and must not
+        # leave the detached kernel running.
+        try:
+            kc = _client(connection_file)
+        except (ValueError, OSError) as e:
+            raise SessionError(f"Kernel wrote an unreadable connection file ({connection_file}): {e}") from e
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)